src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -325,56 +325,65 @@ public void Track(ITelemetry telemetry)
             // It is hidden from intellisense to prevent customer confusion.
             if (this.IsEnabled())
             {
-                string instrumentationKey = this.Context.InstrumentationKey;
+                this.Initialize(telemetry);
 
-                if (string.IsNullOrEmpty(instrumentationKey))
-                {
-                    instrumentationKey = this.configuration.InstrumentationKey;
-                }
+                telemetry.Sanitize();
+
+                this.Channel.Send(telemetry);
 
-                if (string.IsNullOrEmpty(instrumentationKey))
+                if (System.Diagnostics.Debugger.IsAttached)
                 {
-                    return;
+                    this.WriteTelemetryToDebugOutput(telemetry);
                 }
+            }
+        }
 
-                var telemetryWithProperties = telemetry as ISupportProperties;
-                if (telemetryWithProperties != null)
-                {
-                    if (this.Channel.DeveloperMode.HasValue && this.Channel.DeveloperMode.Value)
-                    {
-                        if (!telemetryWithProperties.Properties.ContainsKey("DeveloperMode"))
-                        {
-                            telemetryWithProperties.Properties.Add("DeveloperMode", "true");
-                        }
-                    }
+        /// <summary>
+        /// This method is an internal part of Application Insights infrastructure. Do not call.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Initialize(ITelemetry telemetry)
+        {
+            string instrumentationKey = this.Context.InstrumentationKey;
 
-                    Utils.CopyDictionary(this.Context.Properties, telemetryWithProperties.Properties);
-                }
+            if (string.IsNullOrEmpty(instrumentationKey))
+            {
+                instrumentationKey = this.configuration.InstrumentationKey;
+            }
+
+            if (string.IsNullOrEmpty(instrumentationKey))
+            {
+                return;
```

Comment:
Why does this have to be on TelemetryClient? Wouldn't it make sent to put it something else so as to not clutter the public surface? (You can pass a TelemetryClient in as a parameter)